src/Mirza/SupplyChain/Handlers/UXUtils.hs

```diff
@@ -0,0 +1,60 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE DeriveGeneric         #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TemplateHaskell       #-}
+
+module Mirza.SupplyChain.Handlers.UXUtils
+  (
+    listEventsPretty
+  , PrettyEventResponse (..)
+  ) where
+
+
+import           GHC.Generics                          (Generic)
+
+import           Mirza.Common.GS1BeamOrphans           (LabelEPCUrn (..))
+
+import           Mirza.SupplyChain.Handlers.Queries    (listEventsQuery)
+
+import           Mirza.SupplyChain.ErrorUtils          (throwParseError)
+import           Mirza.SupplyChain.Types               hiding (NewUser (..),
+                                                        User (..))
+import qualified Mirza.SupplyChain.Types               as ST
+
+import qualified Mirza.BusinessRegistry.Types          as BT
+
+import           Data.GS1.DWhat                        (getCompanyPrefix,
+                                                        urn2LabelEPC)
+import qualified Data.GS1.Event                        as Ev
+
+import           Mirza.BusinessRegistry.Client.Servant (uxLocation)
+
+import           Data.Aeson
+import           Data.Aeson.TH
+
+import           Data.Swagger                          (ToSchema (..))
+
+data PrettyEventResponse =
+  PrettyEvent
+  { prettyEvent    :: [Ev.Event]
+  , prettyLocation :: [BT.BusinessAndLocationResponse]
+  } deriving (Show, Eq, Generic)
+$(deriveJSON defaultOptions ''PrettyEventResponse)
```

Comment:
I feel like we should not use generic JSON instances, and instead just define a (very simple) ToJSON instance initially, e.g.

```
{ 
  "eventId": ...,
  "businessName": ...,
  "businessLocation": { "lat": ..., "long"... },
   ...
}
```

Once we have more requirements from the demo app we can expand on the fields. This way we keep the API the demo app is built against isolated from changes to the internal data models.

Reply:
Added a simplified `ToJSON` instance in 97b8ea4.